app/src/main/kotlin/Reversi/ReversiSolverHeuristicSim.kt

```diff
@@ -33,11 +33,11 @@ class ReversiSolverHeuristicSim(initialState: ReversiState)
         }
 
         var depth = 0
-        var currentState = node.state
+        var currenStateType = node.state
```

Comment:
This isn't actually the type. This is actually the state.

Reply:
I'm aware now, the global changer on the IDE, messed this up a bit, will quickly hotfix